src/ctx.rs

```diff
@@ -2,30 +2,100 @@ use crate::include::stdint::uint16_t;
 use crate::include::stdint::uint32_t;
 use crate::include::stdint::uint64_t;
 use crate::include::stdint::uint8_t;
+use std::mem;
+
+pub trait Alias {
+    fn set(&mut self, val: u64);
```

Comment:
Is it a problem that this uses `u64` for all the `val`s?  In the C macros, the integer size would only be expanded to `u64` for the cases where it's actually needed.

Reply:
Also, I'm not really sure we need these `aliasN` types at all.  We can just do a direct `*buf = val`, or `buf[i] = val`, right?  I think C and Rust might have different alignment rules and this is why it was done this way in C, to force the compiler to know that these are aligned writes.  In Rust, `*ptr = val` or `ptr.write()` is always aligned, while `ptr.write_unaligned()` is explicit about being unaligned.

Reply:
I'd like to stick with the `aliasN` approach in this PR, just to keep it as close as possible to the original C. It shouldn't matter too much, though, since you already have improvements lined up.